src/content/docs/essentials/database.mdx

```diff
@@ -0,0 +1,205 @@
+
+```dart
+    await database.into(spendings).insert(
+      SpendingsCompanion.insert(
+        price: Value(3.9),
+        title: 'Coffee',
+        currency: Value('USD'),
+        date: DateTime.now(),
+        receiptImage: Value(Uint8List.fromList([1, 2, 3, 4])),
+      ),
+    );
+```
+
+delete data,
+
+```dart
+    await database.delete(
+        spendings,
+        where: (tbl) => tbl.id.equals(1),
+    ).go();
+```
+
+or update data.
+
+```dart
+    await database.update(
+        spendings,
+        where: (tbl) => tbl.id.equals(1),
+        set: SpendingsCompanion(
+            price: Value(2.9),
+            title: 'Coffee',
+            currency: Value('USD'),
+            date: DateTime.now(),
+            receiptImage: Value(Uint8List.fromList([1, 2, 3, 4])),
+        ),
+    ).go();
+```
+
+More detailed information about data writing operations, you can find  
+[here](https://drift.simonbinder.eu/docs/dart-api/writes/).
+
+### Reading data
+
+To read data, we need to create a query that will return the data we need.
+
+For example, we can retrieve all the data from the table
+
+```dart
+    final allSpendings = await database.select(spendings).get();
+```
+
+Or can filter the data,
+
+```dart
+    final query = await database.select(spendings)
+        ..where((tbl) => tbl.id.equals(1));
+```
+
+sort the data,
+
+```dart
+    final query = await database.select(spendings)
+        ..orderBy(
+            (tbl) => OrderingTerm(expression: tbl.price, mode: OrderingMode.desc),
+        );
+```
+
+or limit the data.
+
+```dart
+    final query = await database.select(spendings)
+        ..limit(10);
+```
+
+No one will stop us from combining everything together, but that's covered
+[here](https://drift.simonbinder.eu/docs/dart-api/select/).
+
+### Migrations
+
+It's often challenging to foresee precisely what data should be in the table right away. In the
+future, new fields or tables may emerge, and for such changes, a migration tool exists.
+
+[Migration API](https://drift.simonbinder.eu/docs/migrations/)
+
+## Executors
+
+In Drift, an executor is a component responsible for handling and executing database queries.
+Drift employs different executors based on the platform to ensure optimal performance and compatibility.
+These platform-specific executors are designed to handle database interactions in a way that aligns
+with the nuances and requirements of each particular platform, providing efficient and seamless
+functionality across various devices.
+
+#### Whats the deal with package?
+
+When moving the database into a separate package with Drift and Flutter, it also simplifies working with
+code generation. The generated code, which creates classes for interacting with the database based on your
+Drift files, becomes part of this standalone package. This contributes to better project organization,
+reduces dependencies, and enhances maintainability.
```

Comment:
The database schema is moved to a different package called `database`. There are a few reasons for this: the schema itself doesn't change frequently, so there is no need to spend time during the codegen. Moreover, it improves project organization as everything related to database is grouped in different package. This package also provides executors logic.

**Executor** is a component tasked with managing and executing database queries. To guarantee optimal performance and compatibility, Drift utilizes different executors tailored to each platform. These platform-specific executors are engineered to align with the unique nuances and requirements of their respective platforms, ensuring efficient and seamless database interactions across a range of devices.